### PR TITLE
[ Application ] Update Logistic Regression to use CCAPI @open sesame 05/16 22:20

### PR DESCRIPTION
**Pull request by jijoongmoon**
This patch includes
 . Update Logistic Regression to use CCAPIs
 . Update Inference process
 . static link about openmp

**Self evaluation:**
1. Build test:	 [X]Passed [ ]Failed [ ]Skipped
2. Run test:	 [X]Passed [ ]Failed [ ]Skipped

Signed-off-by: jijoong.moon <jijoong.moon@samsung.com>

### CODE REVIEW

**Comment by udit01** [comment]
Tested on device. 
Both Training and Inference work.
Using `./tools/package_android.sh` and then using the updated packaged libraries doesn't throw the libomp missing error too.
Thanks for the fix!

File: Applications/LogisticRegression/res/LogisticRegression.ini [hunk 1]
```diff
@@ -1,11 +1,13 @@
...
-Type = Regression	# Network Type : Regression, KNN, NeuralNetwork
-Epochs = 100		# Epochs
+Type = NeuralNetwork	# Network Type : Regression, KNN, NeuralNetwork
+Epochs = 1000		# Epochs
+Learning_rate = 0.001
 Loss = cross    	# Loss function : mse (mean squared error)
                   #                 cross ( cross entropy )
 batch_size = 16		# batch size
 
+
```
Comment (songgot) [inline Applications/LogisticRegression/res/LogisticRegression.ini:10]: How about remove this line?

**Review by songgot — APPROVED** [review]
I add a single comment but LGTM.

File: Applications/LogisticRegression/jni/main.cpp [hunk 2]
```diff
@@ -161,64 +160,79 @@ int main(int argc, char *argv[]) {
...
+        std::vector<float *> answer;
+        float a[1] = {0.0};
+
+        in.push_back(input);
+        l.push_back(label);
+
+        answer = model->inference(1, in, l);
+
+        in.clear();
+        l.clear();
```
Comment (jihochu) [inline Applications/LogisticRegression/jni/main.cpp:226]: Does it needs to be cleared? It seems that *in* and *l* is not used after this.

File: Applications/LogisticRegression/jni/main.cpp [hunk 3]
```diff
@@ -161,64 +160,79 @@ int main(int argc, char *argv[]) {
...
-                        {MAKE_SHARED_TENSOR(nntrainer::Tensor({l}))})[0]
-            ->apply(stepFunction)
-            .getValue(0, 0, 0, 0);
-        std::cout << answer << " : " << l[0] << std::endl;
-        cn += answer == l[0];
+
+        std::vector<float *> in;
+        std::vector<float *> l;
+        std::vector<float *> answer;
+        float a[1] = {0.0};
```
Comment (jihochu) [inline Applications/LogisticRegression/jni/main.cpp:218]: I cannot find code where  *a* is using.

File: Applications/LogisticRegression/res/LogisticRegression.ini (hunk 1 above)
Comment (jijoongmoon) [reply Applications/LogisticRegression/res/LogisticRegression.ini:10]: done

File: Applications/LogisticRegression/jni/main.cpp (hunk 2 above)
Comment (jijoongmoon) [reply Applications/LogisticRegression/jni/main.cpp:226]: there is a loop at L207. So they need to be clear.

File: Applications/LogisticRegression/jni/main.cpp (hunk 3 above)
Comment (jijoongmoon) [reply Applications/LogisticRegression/jni/main.cpp:218]: my bad. removed.

[the remark this answers is not in the document]
Comment (jijoongmoon) [reply Applications/LogisticRegression/jni/main.cpp:218]: same here.

File: Applications/LogisticRegression/jni/main.cpp [hunk 4]
```diff
@@ -161,64 +160,79 @@ int main(int argc, char *argv[]) {
...
+
     for (unsigned int j = 0; j < total_val_data_size; ++j) {
-      nntrainer::Tensor d;
-      std::vector<float> o;
-      std::vector<float> l;
-      o.resize(feature_size);
-      l.resize(1);
 
-      getData(dataFile, o.data(), l.data(), j);
+      float input[3] = {0.0, 0.0, 0.0};
```
[the remark this answers is not in the document]
Comment (jijoongmoon) [reply Applications/LogisticRegression/jni/main.cpp:207]: fixed. thanks.

File: Applications/LogisticRegression/jni/main.cpp [hunk 5]
```diff
@@ -161,64 +160,79 @@ int main(int argc, char *argv[]) {
...
+        float a[1] = {0.0};
+
+        in.push_back(input);
+        l.push_back(label);
+
+        answer = model->inference(1, in, l);
+
+        in.clear();
+        l.clear();
+        if (answer[0][0] < 0.5 && label[0] == 0) {
```
[the remark this answers is not in the document]
Comment (jijoongmoon) [reply Applications/LogisticRegression/jni/main.cpp]: done.

File: Applications/LogisticRegression/jni/main.cpp [hunk 6]
```diff
@@ -161,64 +160,87 @@ int main(int argc, char *argv[]) {
...
     } catch (std::exception &e) {
       std::cerr << "Error during loading weights: " << e.what() << "\n";
       return 1;
     }
     std::ifstream dataFile(data_file);
     int cn = 0;
+    std::vector<float *> in;
+    std::vector<float *> l;
+
+    auto step = [](const float result) {
```
Comment (lhs8928) [inline Applications/LogisticRegression/jni/main.cpp:207]: Now we can delete stepFunction in line 58

Comment (jijoongmoon) [reply Applications/LogisticRegression/jni/main.cpp:207]: Oh. I didn't aware it is there. removed.

**Pull request merged**